### PR TITLE
(chore): GHA - Various

### PR DESCRIPTION
**Pull request by bakerboy448**
Update various github actions and improve general managment

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,39 +1,49 @@
-name: Build mkdocs and deploy to GitHub Pages
+name: Build and Deploy Docs
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
```
Comment (nemchik) [inline .github/workflows/deploy.yml:6]: Don't do this. The existing logic runs mkdocs as a build check to ensure that it passes before merging to master. There was an `if` in the deploy step that only runs the actual deploy when the commits are merged to master.

**Review by nemchik — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (bakerboy448) [reply .github/workflows/deploy.yml:6]: reverted if removal and updated `on` accordingly

**Comment by bakerboy448** [comment]
need some GHA guidance here
- [ ] - Lint - YAML....fails
- [ ] - Lint - markdown, doesn't seem to run?
- [ ] - Lint - editorconfig, doesn't seem to run?

otherwise I think the rest seems fine once those are ironed out

File: .github/workflows/conflicts.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Label Conflicts
+
+on:
+  pull_request:
+    branches:
+      - master
+    types:
+      - opened
+      - synchronize
```
Comment (nemchik) [inline .github/workflows/conflicts.yml:9]:
```suggestion
on: [push, pull_request]
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -1,39 +1,47 @@
...
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
-      - uses: actions/checkout@v3.5.3
-        with:
-          fetch-depth: '0'
-      - uses: actions/setup-python@v4.6.1
+      - name: Checkout repository
+        uses: actions/checkout@v3.5.3
```
Comment (nemchik) [inline .github/workflows/deploy.yml:34]:
```suggestion
        uses: actions/checkout@v3.5.3
        with:
          fetch-depth: '0'
```

File: .github/workflows/labeler.yml [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
-  - pull_request_target
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+    branches:
+      - master
+  push:
+    branches:
+      - master
```
Comment (nemchik) [inline .github/workflows/labeler.yml:12]:
```suggestion
  - pull_request_target
```

File: .github/workflows/lint-editor-json.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+on:
+  push:
+    paths:
+      - "**/*.json"
+      - ".editorconfig"
+  pull_request:
+    paths:
+      - "**/*.json"
+      - ".editorconfig"
```
Comment (nemchik) [inline .github/workflows/lint-editor-json.yml:11]:
```suggestion
[on:
  push:
    paths:
      - "**/*.json"
      - ".editorconfig"
  pull_request:
    paths:
      - "**/*.json"
      - ".editorconfig"](on: [push, pull_request])
```

**Comment by nuxencs** [comment]
just noticed that `- docs/json/sonarr/naming/**` is missing in the `Area: Sonarr` label in the `labeler.yaml`, might want to add it to this PR

**Comment by bakerboy448** [comment]
> just noticed that `- docs/json/sonarr/naming/**` is missing in the `Area: Sonarr` label in the `labeler.yaml`, might want to add it to this PR

https://github.com/TRaSH-Guides/Guides/commit/c5eb359aba97ce3f6c3fb0dab11cf2c88b854c32

**Comment by TRaSH-** [comment]
image
why are we getting this error ?

**Comment by bakerboy448** [comment]
we should be good now. pending iX's review

**Comment by bakerboy448** [comment]
`markdownlint Expected — Waiting for status to be reported` seems to be stuck.  we can ignore for the merge.

**Review by nemchik — APPROVED** [review]

**Pull request merged**